src/Microsoft.AspNet.Http.Abstractions/QueryString.cs

```diff
@@ -148,7 +148,7 @@ public static QueryString Create(IEnumerable<KeyValuePair<string, string>> param
         /// </summary>
         /// <param name="parameters"></param>
         /// <returns>The resulting QueryString</returns>
-        public static QueryString Create(IEnumerable<KeyValuePair<string, StringValues>> parameters)
+        public static QueryString Create(IQueryCollection parameters)
```

Comment:
Does `IQueryCollection` implement `IEnumerable<KeyValuePair<string, StringValues>>`? If so, you can have just one overload that takes the `IEnumerable<...>` and internally have a fast path (non-boxing) for when `parameters` is `IQueryCollection`.